src/main/java/com/mojang/brigadier/ImmutableStringReader.java

```diff
@@ -23,4 +25,8 @@
     char peek();
 
     char peek(int offset);
+
+    boolean isNext(char c);
+
+    boolean isNext(Predicate<Character> predicate);
```

Comment:
What about using [`IntPredicate`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/IntPredicate.html) here? This would avoid boxing and unboxing. But maybe it does not matter much. And it might cause confusion, making users believe this handles code points (including >= `Character.MAX_VALUE`), so this would require renaming the parameter to for example `charPredicate`.